File: base.mvp/src/main/java/alvin/base/mvp/main/MainContracts.java
package alvin.adv.mvp.main;

import android.support.annotation.NonNull;

import java.util.List;

import alvin.adv.mvp.domain.models.NameCard;
import alvin.lib.mvp.contracts.IPresenter;
import alvin.lib.mvp.contracts.IView;

public interface MainContracts {

    interface View extends IView {
        void nameCardsLoaded(@NonNull List<NameCard> nameCards);
    }

    interface Presenter extends IPresenter {
        void loadNameCards();
    }
}

File: ui.ime/src/main/java/alvin/ui/ime/main/MainContracts.kt
package alvin.ui.ime.main

import alvin.lib.mvp.contracts.IPresenter

interface MainContracts {

    interface IView : alvin.lib.mvp.contracts.IView {
        fun showInput(text: String)
        fun hideInputMethod()
    }

    interface Presenter : IPresenter {
        fun textInput(text: String)
    }
}
